src/pages/Router.js

```diff
@@ -0,0 +1,39 @@
+import { render } from '../shared/dom';
+import { Navbar, Footer } from '../components';
+import { HomePage, SearchPage, NotFound } from '.';
+
+const template = `
+    <div id="app-router">
+        <template id="app-navbar"></template>
+        <main>
+            <template id="app-main"></template>
+        </main>
+        <template id="app-footer"></template>
+    </div>
+`;
+
+export const Router = ({ renderOn }) => {
+    render({ html: template, on: renderOn });
+
+    Navbar({ renderOn: '#app-navbar' });
```

Comment:
This shouldn't be here

We have task for preparing the <BaseScreenView />

Which will have a Footer and Navbar, and there's will be a div rendered in which we will injecting pages

Would be great if you could seperate these two logics, @wikbro7650 is handling the BaseView task

Reply:
I don't really understand what's wrong here. I just created a dummy example component. What do you want me to do? I don't think that I should remove those examples because it won't be clear how to work with Router component.

Reply:
1. If anything is temporaty there should be comments with explenation what is what and why
And TODO comments about code which have to be removed in the future

2. My idea is to
- router shouldn't contain any hardcoded HTML or anything like that
-router should be mounted in BaseView
-Router will accept a parametr of array of objects, ex. each object will have a route name which is assigned to, and the component itselft
Based on that we can just pass an array of object and dynamically render each route, without hardcoding like it is now

Something like, you are passing an array, object, whatever you find best, ex. On object

Screens: {
'/home':{
component: HomeScreen
on: '#something'
}
}

Then in router you can just do

Screens[currentRoute].component(...)

And boom, we have dynamic logic, and Route is independent function without any html in it

About the strukture, you can go with object, array, anything that suites you

If you need more explenation or help we can go on a call tomorrow

Reply:
Ok this makes sense. Will do!